base/uk.ac.stfc.isis.ibex.ui.devicescreens.tests/src/uk/ac/stfc/isis/ibex/ui/devicescreens/tests/IconConventionsCheck.java

```diff
@@ -0,0 +1,384 @@
+		// Check whether the current path is for icons or thumbs
+		String iconOrThumbnail = null;
+		if (path.equals(ICONS_PATH)) {
+			iconOrThumbnail = "icon";
+		} else if (path.equals(THUMBS_PATH)) {
+			iconOrThumbnail = "thumbnail";
+		}
+		
+		// Get the OPI types from opi_info, remove duplicates via HashSet
+		HashSet<String> infoTypes = new HashSet<String>(getOPIKeysAndTypes().values());
+		
+		// Convert type names found in opi_info.xml to the icon/thumbnail file names of that type.
+		HashSet<String> infoTypeFiles = new HashSet<String>();
+		for (String type : infoTypes) {
+			infoTypeFiles.add(typeNameToFileName(type, iconOrThumbnail));
+		}
+		
+		List<File> filesUsedByOPIs = new ArrayList<>();
+		// Add the file to the list only if used by an OPI (it exists within opi_info).
+		for (File file : folder.listFiles()) {
+			if (infoTypeFiles.contains(file.getName()) && (!file.isDirectory())) {
+				filesUsedByOPIs.add(file);
+			}
+		}
+
+		return filesUsedByOPIs;
+	}
+	
+	/**
+	 * Returns a map of keys and types from current OPIs in opi_info.xml.
+	 * @return The key and type of current OPIs, empty if an exception occurred while parsing the info file, null if invalid tag name.
+	 */
+	private Map<String, String> getOPIKeysAndTypes() {
+		
+		Map<String, String> keysAndTypes= new HashMap<String, String>();
+		
+		try {
+			File inputFile = new File(OPI_INFO_PATH);
+			DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+			DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
+			Document doc = dBuilder.parse(inputFile);
+			doc.getDocumentElement().normalize();
+			NodeList nList = doc.getElementsByTagName("entry");
+		         
+			for (int temp = 0; temp < nList.getLength(); temp++) {
+				Node nNode = nList.item(temp);
+				if (nNode.getNodeType() == Node.ELEMENT_NODE) {
+					Element eElement = (Element) nNode;
+					String OPIKey = eElement.getElementsByTagName("key").item(0).getTextContent();
+					String OPIType = eElement.getElementsByTagName("type").item(0).getTextContent();
+					keysAndTypes.put(OPIKey, OPIType);
+				}
+			}
+		} catch (Exception e) {
+			e.printStackTrace();
+		}	
+		return keysAndTypes;	
+	}
+
+	
+	/**
+	 * Convert a type name to the name of the file it corresponds to.
+	 * @param typeName The name of the type.
+	 * @param iconOrThumbnail Whether to get the "icon" or the "thumb" file name.
+	 * @return The name of the file.
+	 */
+	public static String typeNameToFileName (String typeName, String iconOrThumbnail) {
+		ComponentType type;
+		
+		// If type is not found as a constant within ComponentType, leave it to UNKNOWN instead.
+		try {
+			type = Enum.valueOf(ComponentType.class, typeName);
+		} catch (IllegalArgumentException e) {
+			type = Enum.valueOf(ComponentType.class, "UNKNOWN");
+		}
+		
+		String filename;
+		if (iconOrThumbnail == "icon") {
+			filename = ComponentIcons.iconNameForType(type);
+		} else if (iconOrThumbnail == "thumb" || iconOrThumbnail == "thumbnail"){
```

Comment:
"thumb", "thumbnail" and "icon" are used a few times, would it be worth pulling them out into constants to avoid typos?